### PR TITLE
Update Documentation

### PR DESCRIPTION
**Pull request by KhoonSun47**
Update Documentation

Changes made:
- DeveloperGuide.md: Update NFR and Glossary, update user stories
and user guides such that it conforms to the glossary
- UserGuide.md: Update Glossary

### CODE REVIEW

**Review by redcolorbicycle — APPROVED** [review]

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -497,46 +497,49 @@ The `DeleteHouseCommandParser` class is used to parse the user input and create
...
+| `* *`     | busy real estate agent      | be able to tell at a glance whether the contact is a buyer or seller | do not need to remember their identity                                    |
+
+[//]: # (@@author )
 
 ### Use cases
 
 (For all use cases below, the **System** is the `EstateEase` and the **Actor** is the `user`, unless specified otherwise)
 
-**Use case: UC01 - Add a home-seller to contact list**
+**Use case: UC01 - Add a seller to contact list**
```
Comment (felixchanyy) [inline docs/DeveloperGuide.md:526]: perhaps should be person list

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -760,11 +770,14 @@ Priorities: Urgent (must-must have) - `* * * *`, High (must have) - `* * *`, Med
 
       Use case ends.
 
+[//]: # (@@author )
+
+[//]: # (@@author lokidoki102)
```
Comment (lokidoki102) [inline docs/DeveloperGuide.md:775]: Dont have to add this empty author tag

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -627,19 +631,22 @@ Priorities: Urgent (must-must have) - `* * * *`, High (must have) - `* * *`, Med
 
       Use case ends.
 
-**Use case: UC05 - Delete a contact**
+[//]: # (@@author )
+
+[//]: # (@@author zengzihui)
```
Comment (lokidoki102) [inline docs/DeveloperGuide.md:636]: same with this empty author tag

**Review by felixchanyy — APPROVED** [review]
LGTM other than the small mistake

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -548,22 +551,22 @@ Priorities: Urgent (must-must have) - `* * * *`, High (must have) - `* * *`, Med
     * 3d1. EstateEase shows an error message. <br>
       Use case ends.
 
-**Use case: UC02 - Add a home-buyer to contact list**
+**Use case: UC02 - Add a buyer to contact list**
```
Comment (felixchanyy) [inline docs/DeveloperGuide.md:554]: as mention above

**Pull request merged**